src/pyhf/__init__.py

```diff
@@ -44,44 +44,69 @@ def set_backend(backend, custom_optimizer=None):
         >>> pyhf.set_backend("tensorflow")
         >>> pyhf.tensorlib.name
         'tensorflow'
-        >>> pyhf.set_backend(b"pytorch")
+        >>> pyhf.tensorlib.precision
+        '32b'
+        >>> pyhf.set_backend(b"pytorch", precision="64b")
         >>> pyhf.tensorlib.name
         'pytorch'
+        >>> pyhf.tensorlib.precision
+        '64b'
         >>> pyhf.set_backend(pyhf.tensor.numpy_backend())
         >>> pyhf.tensorlib.name
         'numpy'
+        >>> pyhf.tensorlib.precision
+        '64b'
 
     Args:
         backend (`str` or `pyhf.tensor` backend): One of the supported pyhf backends: NumPy, TensorFlow, PyTorch, and JAX
         custom_optimizer (`pyhf.optimize` optimizer): Optional custom optimizer defined by the user
+        precision (`str`): Floating point precision to use in the backend: ``64b`` or ``32b``. Default is backend dependent.
 
     Returns:
         None
     """
     global tensorlib
     global optimizer
 
+    _valid_precisions = ["32b", "64b"]
+    backend_kwargs = {}
+
+    if isinstance(precision, (str, bytes)):
+        if isinstance(precision, bytes):
+            precision = precision.decode("utf-8")
+        precision = precision.lower()
+
     if isinstance(backend, (str, bytes)):
         if isinstance(backend, bytes):
             backend = backend.decode("utf-8")
         backend = backend.lower()
+
+        if precision is not None:
+            backend_kwargs["precision"] = precision
+
         try:
-            backend = getattr(tensor, "{0:s}_backend".format(backend))()
+            backend = getattr(tensor, f"{backend:s}_backend")(**backend_kwargs)
         except TypeError:
             raise InvalidBackend(
-                "The backend provided is not supported: {0:s}. Select from one of the supported backends: numpy, tensorflow, pytorch".format(
-                    backend
-                )
+                f"The backend provided is not supported: {backend:s}. Select from one of the supported backends: numpy, tensorflow, pytorch"
             )
 
-    _name_supported = getattr(tensor, "{0:s}_backend".format(backend.name))
+    _name_supported = getattr(tensor, f"{backend.name:s}_backend")
     if _name_supported:
         if not isinstance(backend, _name_supported):
             raise AttributeError(
-                "'{0:s}' is not a valid name attribute for backend type {1}\n                 Custom backends must have names unique from supported backends".format(
-                    backend.name, type(backend)
-                )
+                f"'{backend.name:s}' is not a valid name attribute for backend type {type(backend)}\n                 Custom backends must have names unique from supported backends"
             )
+    if backend.precision not in _valid_precisions:
```

Comment:
what is this check for? We only care about the `backend.precision` if a user is using one of our backends and tries to set it... but if it's a custom backend(?) -- then why do we care what precision they use?

Reply:
What happens if they typo the precision or try to set nonsense?

Reply:
@kratsg Or are you just saying that this should get moved in a level so it only gets hit for supported backends and let them deal with it if they make their own?

Reply:
Ahh, I see.. our backends don't quite protect against this...

https://github.com/scikit-hep/pyhf/blob/161d23ac0d892e0361135326f5c1cac4c4620134/src/pyhf/tensor/numpy_backend.py#L41-L49

The thing is our `set_backend` would fail if someone created a backend that handled 128b or something. Valid precision should be up to the backends to define. We don't necessarily use that precision information anywhere except to set up the dtypemap or to uniquely identify backends.

Reply:
So you would vote to just remote the precision checks entirely? Or only check for our supported backends.

Reply:
That's the key question isn't it? I am not sure what the best thing to do here is. Presumably, we should have these precision checks at the backend level, rather than at the pyhf level _but_ I am fine with this for now, since we don't necessarily have custom backends being supported and architectures likely won't be anything but 32/64...

Reply:
Yeah, fair. I've added this for followup to Issue #981.